spec/models/miq_report_result_spec.rb

```diff
@@ -1,14 +1,56 @@
 describe MiqReportResult do
-  context "persisting generated report results" do
-    before(:each) do
-      EvmSpecHelper.local_miq_server
+  before :each do
+    EvmSpecHelper.local_miq_server
+
+    @user1 = FactoryGirl.create(:user_with_group)
+  end
+
+  context "report result created by User 2 with current group 2" do
+    before :each do
+      @report = FactoryGirl.create(:miq_report)
+      group_2 = FactoryGirl.create(:miq_group)
+      @user1.miq_groups << group_2
+      @report_result1 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report.id, :miq_group => group_2)
+      @report_result2 = FactoryGirl.create(:miq_report_result, :miq_report_id => 999_999_999)
+      User.current_user = @user1
+    end
+
+    describe ".with_report" do
+      it "returns only requested report results" do
+        report_result = MiqReportResult.with_report(@report_result1.miq_report_id)
+        expect(report_result).to include(@report_result1)
+      end
+
+      it "returns all requested report results" do
+        report_result = MiqReportResult.with_report
+        expect(report_result).to include(@report_result1)
+        expect(report_result).to include(@report_result2)
```

Comment:
@lpichler Maybe use match_array here to make sure the arrays [@report_result1, @report_result2] match, order independently